enumerables.rb

```diff
@@ -0,0 +1,155 @@
+module Enumerable
+    def my_each
+        for i in 0...self.length do
+            yield(self[i])
+        end
+    end
+    
+    def my_each_with_index
+        for i in 0...self.length do
+            yield(self[i], i)
+        end
+    end
+
+    def my_select
+        a = []
+        my_each do |n|
+            if yield(n)
+                a.push(n)
+            end
+        end
+        a
+    end
+
+    def my_all?
+        my_each do |n|
+            if !yield(n)
+                return false
+            end
+        end
+        true
+    end
+
+    def my_none?
+        my_each do |n|
+            if yield(n)
+                return false
+            end
+        end
+        true
+    end
+
+    def my_count(arg = nil)
+            count = 0
+            if arg == nil && block_given? == false
+            return self.length    
+            elsif block_given?
+                my_each do |n|
+                    if yield(n)
+                    count += 1
+                    end
+                end
+                return count
+            else
+            my_each do |n|
+                if arg == n
+                count += 1
+                end
+            end
+            return count
+        end  
+    end
+
+    def my_map(&block)
+        newarr = []
+        my_each_with_index do | n, i |
+            if block == true
+            newarr[i] = block.call(n)
+            else
+            newarr[i] = yield(n)
+            end
+        end
+        return newarr
+    end
+
```

Comment:
- [ ]  my_map should return a new array with the results of running a given block once for every element in enum::range

- [ ] enumerables #my_map returns an Enumerator if no block is given